src/dialog/screen_enemies.cpp

```diff
@@ -0,0 +1,260 @@
+					Button(
+						text = Map.CanPaste() ? fmt::format("Past&e (from {}x{:02x})", (Map.CopyScr() >> 8) + 1, Map.CopyScr() & 255) : "Past&e from screen",
+						colSpan=2, fitParent=true,
+						minwidth = 90_px,
+						onClick = message::OK,
+						disabled = !Map.CanPaste()
+					),
+					Button(
+						text = "&Flags",
+						colSpan = 2, fitParent = true,
+						minwidth = 90_px,
+						onClick = message::FLAGS
+					),
+					Button(
+						text = "&Pattern",
+						colSpan = 2, fitParent = true,
+						minwidth = 90_px,
+						onClick = message::PATTERN
+					)
+				),
+				Row(
+					vAlign = 1.0,
+					topPadding = 0.5_em,
+					spacing = 1_em,
+					Button(
+						text = "O&K",
+						fitParent = true,
+						minwidth = 90_px,
+						onClick = message::OK
+					),
+					Button(
+						text = "Cancel",
+						fitParent = true,
+						minwidth = 90_px,
+						onClick = message::CANCEL
+					)
+				)
+			),
+			Column(
+				widgPrev = TileFrame(visible = false),
+				widgInfo = Label(text = "")
+			)
+		)
+	);	
+	RebuildList();
+	UpdatePreview();
+	return window;
+}
+
+bool ScreenEnemiesDialog::handleMessage(const GUI::DialogMessage<message>& msg)
+{
+	bool refresh = false;
+	switch (msg.message)
+	{
+	case message::COPY:
+		copied_enemy_id = thescr->enemy[scr_enemies->getSelectedIndex()];
+		UpdatePreview();
+		break;
+	case message::PASTE:
+		if (copied_enemy_id > 0 && copied_enemy_id < eMAXGUYS)
+			thescr->enemy[scr_enemies->getSelectedIndex()] = copied_enemy_id;
+		RebuildList();
+		UpdatePreview();
+		refresh = true;
+		break;
+	case message::EDIT:
+	{
+		int32_t current_enemy = thescr->enemy[scr_enemies->getSelectedIndex()];
+		EnemyListerDialog(current_enemy, true).show();
+		if (lister_sel_val != current_enemy)
+		{
+			thescr->enemy[scr_enemies->getSelectedIndex()] = lister_sel_val;
+			RebuildList();
+			UpdatePreview();
+		}
+		refresh = true;
+		break;
+	}
+	case message::CLEAR:
+		thescr->enemy[scr_enemies->getSelectedIndex()] = 0;
+		RebuildList();
+		UpdatePreview();
+		refresh = true;
+		break;
+	case message::PASTEFROMSCREEN:
+		Map.DoPasteScreenCommand(PasteCommandType::ScreenEnemies);
+		RebuildList();
+		UpdatePreview();
+		refresh = true;
+		break;
+	case message::FLAGS:
+		call_screendata_dialog(3);
+		refresh = true;
+		break;
+	case message::PATTERN:
+		call_enemypattern_dialog();
+		refresh = true;
+		break;
+	case message::OK:
+	{
+		bool end = false;
+		bool confirm = true;
+		for (int32_t i = 0; i < 10; i++)
+		{
+			if (thescr->enemy[i] == 0)
+				end = true;
+			else if (end)
+			{
+				AlertDialog("Inactive Enemies!",
+					"Enemies won't appear if they're preceded by '(None)' in the list! Continue?",
+					[&](bool ret, bool)
+					{
+						confirm = ret;
+					}).show();
+			}
+		}
+		if (!confirm) rerun_dlg = true;
+		return true;
+	}
+	case message::CANCEL:
+		memcpy(thescr->enemy, oldenemy, 10 * sizeof(word));
+		return true;
+	}
+	rerun_dlg = refresh;
```

Comment:
```suggestion
	if(refresh) rerun_dlg = true;
```
to match the `if(!confirm) rerun_dlg = true;` above; and also probably cleaner in general to not set rerun_dlg to anything but `true`, as nothing ever should be writing it to false except the dialog runner code (in this case, it should always ALREADY be false when you write it to false, so like, this *really* doesn't matter that much.